eth-contracts/contracts/ClaimsManager.sol

```diff
@@ -231,7 +232,10 @@ contract ClaimsManager is RegistryContract {
         uint lastClaimedForSP = Staking(
             registry.getContract(stakingProxyOwnerKey)
         ).lastClaimedFor(_sp);
-        return (lastClaimedForSP < currentRound.fundBlock);
+        (,,,uint numEndpoints,,) = ServiceProviderFactory(
+            registry.getContract(serviceProviderFactoryKey)
+        ).getServiceProviderDetails(_sp);
+        return (lastClaimedForSP < currentRound.fundBlock && numEndpoints > 0);
```

Comment:
is this because SPFactory now checks claimPending? and we explicitly don't want to check that the within bounds is met because if they falal out of bounds it would never let them call register etc?

Reply:
yes that's right and in the case that a round has been completed and a new SP is incoming, without this number of endpoints check they would not be allowed to claim

Reply:
we should adad a comment above this function outlining this case. my first reaction was why would it check endpoints but not bounds, but then i saw SPFactory and i understood why

Reply:
added